railties/test/application/configuration_test.rb

```diff
@@ -2745,21 +2745,31 @@ def index
 
     test "Rails.application.config.action_mailer.smtp_settings have open_timeout and read_timeout defined as 5 in 7.0 defaults" do
       remove_from_config '.*config\.load_defaults.*\n'
-      add_to_config 'config.load_defaults "7.0"'
+      add_to_config <<-RUBY
+        config.action_mailer.smtp_settings = { domain: "example.com" }
+        config.load_defaults "7.0"
+      RUBY
 
       app "development"
 
-      assert_equal 5, ActionMailer::Base.smtp_settings[:open_timeout]
-      assert_equal 5, ActionMailer::Base.smtp_settings[:read_timeout]
+      smtp_settings = { domain: "example.com", open_timeout: 5, read_timeout: 5 }
+
+      assert_equal smtp_settings, ActionMailer::Base.smtp_settings
+      assert_equal smtp_settings, Rails.configuration.action_mailer.smtp_settings
```

Comment:
This should only contain `domain` to keep the same behavior.

Reply:
On 6.1 smtp_settings is passed down directly into ActionMailer from the config, and it's the same object. That's why I added this line. Maybe I'm missing something? 

```
Loading development environment (Rails 6.1.4.4)
irb(main):001:0> Rails.configuration.action_mailer.smtp_settings
=> {:domain=>"example.com"}
irb(main):002:0> ActionMailer::Base.smtp_settings
=> {:domain=>"example.com"}
irb(main):003:0> ActionMailer::Base.smtp_settings.object_id
=> 12100
irb(main):004:0> Rails.configuration.action_mailer.smtp_settings.object_id
=> 12100
irb(main):005:0> Rails.configuration.action_mailer.smtp_settings["foo"] = "bar"
=> "bar"
irb(main):006:0> ActionMailer::Base.smtp_settings
=> {:domain=>"example.com", "foo"=>"bar"}
```

Reply:
hmmm, I see.